### ISSUE TITLE
 Missing scrollbar on DB Schema viewlet

### ISSUE THREAD

**Issue by tlhackque** [sqlitebrowser/sqlitebrowser#1450]
Details for the issue
--------------------

#### What did you do?
Opened "DB Schema" tab on right pane viewlet

#### What did you expect to see?
A scrolbar

#### What did you see instead?

Truncation.

A vertical scrollbar appears when necessary.

However, a horizontal scrollbar does not.

This is limiting when you have a schema with interesting definitions - check constraints, foreign keys - even long variable names.  And pretty much any CREATE TABLE line gets truncated at the right, no matter how trivial.

A horizontal scrollbar should appear as needed - or, since i think that in non-trivial cases, that's pretty much always, it can probably be there unconditionally, if that's easier...

Useful extra information
-------------------------

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

- [ x] Windows: ( _version:_ 10 pro___ )
- [ ] Linux: ( _distro:_ ___ )
- [ ] Mac OS: ( _version:_ ___ )
- [ ] Other: ___

#### What is your DB4S version?

- [ ] 3.10.1
- [ ] 3.10.0
- [ ] 3.9.1
- [x ] Other: _3.10.99 24-Jun__

#### Did you also

- [x ] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [ ] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by justinclift** [comment]
Would you be ok to screenshot the scrollbar not appearing for you?  Just tried it out here on my CentOS 7 desktop, and the horizontal scrollbar is showing up.

**Comment by tlhackque** [comment]
Windows.

![image](https://user-images.githubusercontent.com/332018/41824382-19f16dd4-77dd-11e8-8d31-ae49d5e5c270.png)

**Comment by justinclift** [comment]
Oh.. yeah... actually it's not real great for me either now that I try it out a bit more.

For example, when using the [Assembly Election 2017](https://dbhub.io/nicva/Assembly%20Election%202017.sqlite) example data, the horizontal scrollbar doesn't show up at all unless I reduce the width of the pane drastically.

![db4s-db-schema1](https://user-images.githubusercontent.com/406299/41824420-86b5e75e-7807-11e8-9a17-232d3b806406.png)
![db4s-db-schema2](https://user-images.githubusercontent.com/406299/41824422-8acfd4e4-7807-11e8-9d89-bf02a172cdef.png)

Maybe it's miscalculating how wide the "optimal" view width should be?

**Comment by tlhackque** [comment]
Yes - if you replace "optimal" with "necessary" - that's usually the widget's job.  It may not be getting an event indicating that more data has been added, or a resize, or perhaps something is overlapping the widget so it has less space than it thinks, or ... There are a lot of edge cases in GUI internals.  Which is why I leave the details to the experts.

What may be happening here is that the schema display is a subwindow of the larger panel, which is again a subwindow of the the main window.  So perhaps the window placement manger is using the wrong subwindow to calculate width.  You might see if the behavior changes depending on space allocated to the parent (or north peer) window.  (A guess - I know enough to be dangerous :-)

"It's not how well the elephant dances, it's that he dances at all"

**Comment by chrisjlocke** [comment]
> It may not be getting an event indicating that more data has been added, or a resize, or perhaps something is overlapping the widget 

I think it might be simpler than that, and is just the widths of the columns.  If the widths of the columns fit, then there is no scrollbar to display.

See this screencast: https://screencast-o-matic.com/watch/cF1TD2Fqie

So there is no scrollbar, but if I move the mouse to the last column, the cursor changes, and the column can be extended.  Double clicking at the point where the mouse cursor changes (ie, the border of the cell, if it were visible) extends the column to fit the contents - ie, maximum size.  Double clicking the other columns does the same thing, but obviously shrinks them down, as they are initially larger than their contents.
Finally, I end the screencast by moving the column width down to 'enough' where the scrollbar disappears.

So in summary, its not a widget issue (directly), but it just needs the column expanding to take into account its contents.  An easy workaround.

**Comment by mgrojo** [comment]
@chrisjlocke It's exactly as you describe. Nevertheless, it can be improved so it adjusts to the content, as the same column in the Database Structure tab does.

**Comment by mgrojo** [comment]
Good news. It should be solved in tomorrow's nightly build.

**Comment by chrisjlocke** [comment]
It works, but as a suggestion, I'd also resize column 1 (name) and column 2 (type).  By default, the widths are quite wide for a small docked window.

![image](https://user-images.githubusercontent.com/7872218/42081433-55aa05cc-7b7d-11e8-9e12-76932e13cbd2.png)

Just a thought...

**Comment by mgrojo** [comment]
@chrisjlocke I've also resize to (visible) content the Name schema. The Type schema has problems, since there isn't any visible content when the tree is collapsed, and consequently the column width is set to the header width, which is lower than the needed for the field types when the tables are expanded. Resizing Name seems enough, nevertheless.

**Comment by chrisjlocke** [comment]
I bow to your greatness.  😄
